src/main/java/de/tum/in/www1/artemis/service/connectors/bamboo/BambooMigrationService.java

```diff
@@ -0,0 +1,147 @@
+
+/**
+ * Services for executing migration tasks for Bamboo.
+ */
+@Service
+@Profile("bamboo")
+public class BambooMigrationService implements CIMigrationService {
+
+    @Value("${artemis.continuous-integration.url}")
+    protected URL bambooServerUrl;
+
+    @Value("${server.url}")
+    private String artemisServerUrl;
+
+    private final RestTemplate restTemplate;
+
+    public BambooMigrationService(@Qualifier("bambooRestTemplate") RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
+
+    @Override
+    public void overrideBuildPlanNotification(String projectKey, String buildPlanKey, VcsRepositoryUrl vcsRepositoryUrl) {
+        List<Long> notificationIds = getAllArtemisBuildPlanServerNotificationIds(buildPlanKey);
+
+        for (var id : notificationIds) {
+            deleteBuildPlanServerNotificationId(buildPlanKey, id);
+        }
+
+        createBuildPlanServerNotification(buildPlanKey, artemisServerUrl + NEW_RESULT_RESOURCE_API_PATH);
+    }
+
+    /**
+     * Returns a list of all notification ids for the given build plan for this server.
+     * Bamboo doesn't provide a REST endpoint for this, so we have to parse the HTML page using the admin chain API.
+     *
+     * @param buildPlanKey The key of the build plan, which is usually the name combined with the project, e.g. 'EIST16W1-GA56HUR'.
+     * @return a list of all notification ids
+     */
+    private List<Long> getAllArtemisBuildPlanServerNotificationIds(String buildPlanKey) {
+        MultiValueMap<String, String> parameters = new LinkedMultiValueMap<>();
+        parameters.add("buildKey", buildPlanKey);
+        String requestUrl = bambooServerUrl + "/chain/admin/config/defaultChainNotification.action";
+        UriComponentsBuilder builder = UriComponentsBuilder.fromUriString(requestUrl).queryParams(parameters);
+
+        var response = restTemplate.exchange(builder.build().toUri(), HttpMethod.GET, null, String.class);
+        var html = response.getBody();
+        if (html == null) {
+            return List.of();
+        }
+        Element notificationTableBody = Jsoup.parse(html).selectFirst("table#notificationTable tbody");
+        if (notificationTableBody == null) {
+            return List.of();
+        }
+        // First column is the event, second column the recipient, third the actions
+        // If there is a URL, the URL is the recipient. In that case we take the notification id from the edit button
+        Elements entries = notificationTableBody.select("tr");
+        List<Long> notificationIds = new ArrayList<>();
+        for (Element entry : entries) {
+            Elements columns = entry.select("td");
+            if (columns.size() != 3) {
+                continue;
+            }
+            String recipient = columns.get(1).text();
+            String actions = columns.get(2).toString();
+            Pattern editNotificationIdPattern = Pattern.compile(".*?id=\"editNotification:(\\d+)\".*?");
```

Comment:
I don’t know how many entries are to be expected in the table. If there are many, I’d move the compile out of the loop since that is an comparatively expensive operation 
(still unlikely to be significant compared to networking stuff here unless _a lot_ of `tr`s are expected).